enhancements/observability/multicluster-observability-addon-interoperability.md

```diff
@@ -0,0 +1,217 @@
+
+* As a fleet administrator, I want to export all telemetry signals collected by MCOA to an OTLP compatible endpoint(s).
+* As a fleet administrator, I want to filter sensitive data before it is exported to MCO telemetry store or 3rd party OTLP endpoint.
+* As a fleet administrator, I want to decide which data is exported to MCO telemetry store or 3rd party OTLP endpoint.
+
+### Goals
+
+* Use a single protocol (OTLP) for exporting all data to MCO and/or 3rd party system.
+* Provide a single configuration API on MCO CRD for exporting all telemetry data.
+* Provide unified filtering and routing capabilities for all exported telemetry data.
+
+### Non-Goals
+
+* Data visualization and querying.
+
+## Proposal
+
+The following sections describe in detail the required resources as well as the workflow to enable collection and forwarding of logs and traces on an RHACM managed fleet of OCP clusters.
+
+The following section describes how data exportation, routing and filtering is 
+configured in MCO and MCOA.
+
+![Architecture](./multicluster-observability-addon-interoperability-arch.jpg)
+
+### Workflow Description
+
+1. Configure OTLP endpoint in MCO (`MultiClusterObservability`) CR.
+2. The MCOA configures an additional OTLP exporter in the OpenTelemetry collector. The 
+   exporter is in the pipeline that receives all data.
+3. (optional) Filtering (e.g. for PII) can be configured in 
+   `OpenTelemetryCollector` CR manged by MCOA by [transformprocessor](https://github.
+   com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/transformprocessor/README.md).
+4. (optional) Routing can be configured in `OpenTelemetryCollector` CR managed by MCOA by
+   [routingprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/routingprocessor).
+
+
+### API Extensions
+
+None - no new APIs for CRDs are introduced.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### General configuration and fleet-wide stanzas
+
+To support above workflow MCOA deploys additional collector which forwards all data to 
+MCO telemetry store and/or 3rd party OTLP endpoint.
+
+- An `OpenTelemetryCollector` resource that enables receivers for all telemetry 
+  signals and an OTLP exporter for and forwarding to the MCO store and 3rd party 
+  vendor.
+
+#### Hypershift [optional]
+
+N/A
+
+### Drawbacks
+
+- MCOA configuration through the MultiClusterObservability: the MCO CR nowadays has an already extensive set of configuration fields, when designing the MCOA configuration, we will need to take extra caution as to not make this CR more complex and hard to navigate;
+- MCOA manifest sync: with MCOA being deployed by MCO we will need to set up a procedure to maintain the MCOA manifests that live in the MCO repo up to date.
+- CRD conflicts: MCOA will leverage the CRDs from other operators we will have to ensure that we will not be running into situations where two operators are managing the same CRD
```

Comment:
To add to this,

- Native OTLP ingestion is not stable for upstream projects like Prometheus and Thanos, and even if they work, they are not yet performant enough to replace remote write, and there are quite some decisions to be taken, on how to handle differences between the protocol.
- OTLP remote write exporter can be used but then this is also a slower path as compared to directly remote writing metrics, and such conversions from one protocol to another can result in broken semantics at times.

Reply:
I think in this proposal @saswatamcode, there is a collector running in the hub that translates back to remote_write to avoid the drawbacks you mentioned.

Reply:
Yup either way we go, there are drawbacks.

Reply:
> I think in this proposal @saswatamcode, there is a collector running in the hub that translates back to remote_write to avoid the drawbacks you mentioned.

No, this proposal does not imply running a collector on the hub that translates to protocols supported by the stores.

Reply:
@saswatamcode 

> Native OTLP ingestion is not stable for upstream projects like Prometheus and Thanos, and even if they work, they are not yet performant enough to replace remote write

Are there available benchmark results to reason about this?

Reply:
@pavolloffay it is a translation layer built into Prometheus iiuc. There are a variety of discussions and issues around performance and supporting otel semantics in this milestone: https://github.com/prometheus/prometheus/milestone/12 
And all of this translates into Thanos as well, as we are a sort of downstream consumer of Prometheus code.

Reply:
Thanks for the link. The translator most likely incurs some CPU/mem cost.  Are there some benchmarks that show that or even a change in the ingestion throughput?

Reply:
Between remote write vs native otlp? I don't think there is anything publicly available as a benchmark, maybe some prombench scenario. But the way this translator works is by translating otlp to prometheus remote write requests https://github.com/prometheus/prometheus/blob/main/storage/remote/otlptranslator/prometheusremotewrite/metrics_to_prw.go#L41 so it is doing additional work on top of regular remote write ingestion.

To get around this vendors like grafana have products like https://grafana.com/docs/grafana-cloud/send-data/alloy/

Hopefully over some time we see it become more native in Prometheus, and equally performant 🙂

Reply:
> To get around this vendors like grafana have products like https://grafana.com/docs/grafana-cloud/send-data/alloy/

That is just a custom build like OTEL collector. We have Red Hat build of OpenTelemetry  https://docs.openshift.com/container-platform/4.15/observability/otel/otel-configuration-of-otel-collector.html. In the next version we will add Prometheus Remote Write exporter.

Reply:
Here are some benchmark results, comparing
1. Prometheus -> RW -> RW endpoint
2. OTelCol -> OLTP -> OTLP endpoint
3. OtelCol -> RW -> RW endpoint

https://github.com/danielm0hr/edge-metrics-measurements/blob/main/talks/DanielMohr_PromAgentVsOtelCol.pdf

I guess in scenario 3 this translation is happening, right?